programs/protocol-contracts-solana/src/lib.rs

```diff
@@ -367,73 +364,30 @@ pub mod gateway {
             Errors::SPLAtaAndMintAddressMismatch,
         );
 
-        // test whether the recipient_ata is created or not; if not, create it
+        // test whether the recipient_ata is created or not; if not log and exit successfully
```

Comment:
Can we explain why we log and exit successfully instead of making the tx fails

Reply:
@lumtis the core issue here is the possibility of a "legitimate failed withdraw tx" (for a lack of better work) which means the withdraw cannot be accomplished and we tried, as opposed to other "illegitimate failed withdraw" which may include: wrong/fake TSS signature, replay using a wrong nonce). 

For "legitimate failed withdraw tx", zetaclient should report as "revert" like in Ethereum case so the solana nonce on zetacore will increment and we move on. **However if the Solana tx actually failed, the program nonce will not increment**. In this case we need to allow the solana tx to succeed, but without transfering SPL tokens and indicate in logs that it "reverted". 

For "illegitimate failed withdraw", the correct behavior is that zetaclients would ignore such tx as they are not authorized. The solana program simply fail them.

Reply:
I see, I think still make sense to precise this in the comment